message ClusterConfiguration {

    required string version = 1;

}

message TableConfiguration {

    // The relation name of this table.
    required string name = 1;

    // The columns that will be monitored on this table.
    repeated string columns = 2;

    // The schema where the table is located.
    optional string schema = 3 [default="public"];

}

message DatabaseConfiguration {

    // A libpq connection string, as defined by http://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING
    required string dsn = 1;

}

message ReplicationSetConfiguration {

    // The databases where this replication setresides.
    repeated DatabaseConfiguration databases = 1;

    // The tables within this replication set.
    repeated TableConfiguration tables = 2;

}
